main.tf

```
@@ -1,15 +1,32 @@
 locals {
-  connect_vcs_repo = var.repository_identifier != null ? { create = true } : {}
+  connect_vcs_repo     = var.repository_identifier != null ? { create = true } : {}
+  tfc_agent_accountid  = var.tfc_agent_role_configuration["AccountId"]
+  tfc_agent_externalid = var.tfc_agent_role_configuration["ExternalId"]
 }
 
+// Add a flag to specify auth method, switch between user and role for TFC
+
 module "workspace_account" {
```

We should update this module name too to be consistent:

```suggestion
module "workspace_iam_user" {
```

We can avoid breaking people's setup by using a moved block in a `refactor.tf`:

```hcl
moved {
  from = module.workspace_account
  to   = module.workspace_iam_user
}
```

If not familiar with it there are some examples [here](https://developer.hashicorp.com/terraform/language/modules/develop/refactoring).